packages/kbn-plugin-generator/sao_template/template/public/types.ts

```diff
@@ -0,0 +1,11 @@
+import { NavigationPublicPluginStart } from '<%= relRoot %>/src/plugins/navigation/public';
+
+export interface <%= upperCamelCaseName %>PluginSetup {
+  getGreeting: () => string;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface <%= upperCamelCaseName %>PluginStart {}
+
+export interface AppPluginDependencies { 
+  navigation: NavigationPublicPluginStart 
+};
```

Comment:
- `AppPluginDependencies` -> `AppPluginStartDependencies` (as there could also be a `AppPluginSetupDependencies` for `setup` phase)

- Should probably be `<%= upperCamelCaseName %>PluginStartDependencies`